### PR TITLE
Fix libnvinfer conflict

### PR DESCRIPTION
**Pull request by EmmaQiaoCh**
1. remove the specified version of libnvinfer
2. copy libcupti.so.12 instead of libcupti.so.11.8 which is not in upstream container
3. update TFDE_VER=v0.3 to fix some build error

### CODE REVIEW

**Comment by jperez999** [comment]
Running this in pre-release environment. Will report back if any issues... otherwise I will approve.

File: docker/dockerfile.merlin [hunk 1]
```diff
@@ -198,15 +198,15 @@ RUN apt update -y --fix-missing && \
         #   Required to build Protocol Buffers.
         autoconf automake libtool \
         #   Required to build Hadoop.
-        default-jdk maven pkg-config \
```
Comment (jperez999) [inline docker/dockerfile.merlin:201]: Love that we can remove this.

**Comment by karlhigley** [comment]
The test failure here is unrelated and we have an open PR to fix it. Will merge that one and re-run this.

**Comment by EmmaQiaoCh** [comment]
@jperez999 @karlhigley Thanks for your info, but we just hit regression errors when testing SOK and hps_trt on the new TF, so please hold on this PR although your tests passed. We may need more changes.

**Review by karlhigley — APPROVED** [review]

**Pull request merged**